config/admin-functions/expand-var-filesystem.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Using a flag file to skip resize operations if already completed
+# There's not an issue running these commands again, but this saves time
+flag_file="/home/VAR_RESIZED"
+
+if [[ -f ${flag_file} ]]; then
+  exit 0
+fi
+
+# Find the mapped device for /var, need this to look up the LV later
+var_map=$( df -l /var | grep mapper | awk '{print $1}' )
+
+# This identifies the PV used in our config. It assumes our Vx naming convention
+pvs_path=$( pvs | grep Vx | awk '{print $1}' )
+
+# Using the PV path, get the parent device path we'll need for resizing
+parent_partition=$( lsblk -ndo pkname ${pvs_path} )
+
+volume_to_extend="NONE"
+
+# Deteremine the volume to extend based on whether encrypted /var is used or not
+if [[ $var_map =~ "var_decrypted" ]]; then
+  volume_to_extend=$( grep var_decrypted /etc/crypttab | awk '{print $2}' )
+else
+  volume_to_extend=$( lvdisplay ${var_map} | grep "LV Path" | awk '{print $3}' )
+fi
+
+# Vx convention: we always expect partition 3
+# if this changes in the future, we'll need to add logic to detect
+growpart "/dev/${parent_partition}" 3 || true
```

Comment:
is this assumption necessary? Can we find the partition number from the label? I can imagine this being a silent failure if we expand the wrong partition and we don't find out until scale testing that `/var` is tiny.

Reply:
I didn't see an easy/obvious way to do this, but I also didn't spend too much time on it since our preseeds for both vxdev and production result in a partition scheme where `/boot/efi` is on 1, `/boot` is on 2, and everything else is on 3. I can spend some more time on it though to see if there's a good approach that will work across our different system types.

Reply:
hmmmm, okay, you make a convincing point that this is likely to be quite stable unless we make a major change. I'm good with it.